rust/src/builders/tx_builder.rs

```diff
@@ -740,121 +599,35 @@ impl TransactionBuilder {
         Ok(())
     }
 
-    /// Set explicit Mint object and the required witnesses to this builder
-    /// it will replace any previously existing mint and mint scripts
-    /// NOTE! Error will be returned in case a mint policy does not have a matching script
-    pub fn set_mint(&mut self, mint: &Mint, mint_scripts: &NativeScripts) -> Result<(), JsError> {
-        assert_required_mint_scripts(mint, Some(mint_scripts))?;
-        self.mint = Some(mint.clone());
-        self.set_native_scripts(mint_scripts);
-        Ok(())
-    }
-
-    /// Returns a copy of the current mint state in the builder
-    pub fn get_mint(&self) -> Option<Mint> {
-        self.mint.clone()
-    }
-
-    /// Returns a copy of the current witness native scripts in the builder
-    pub fn get_native_scripts(&self) -> Option<NativeScripts> {
-        self.native_scripts.clone()
-    }
-
-    fn _set_mint_asset(&mut self, policy_id: &PolicyID, policy_script: &NativeScript, mint_assets: &MintAssets) {
-        let mut mint = self.mint.as_ref().cloned().unwrap_or_else(Mint::new);
-        let is_new_policy = mint.insert(policy_id, mint_assets).is_none();
-        let mint_scripts = {
-            let mut witness_scripts = self.native_scripts.as_ref().cloned()
-                .unwrap_or_else(NativeScripts::new);
-            if is_new_policy {
-                // If policy has not been encountered before - insert the script into witnesses
-                witness_scripts.add(&policy_script.clone());
+    pub fn add_mint(&mut self, result: &MintBuilderResult) {
+        if let Some(ref data) = result.aggregate_witness {
+            self.witness_set_builder.add_input_aggregate_witness_data(data);
+            if let InputAggregateWitnessData::PlutusScript(witness, _, _) = data {
+                let redeemer = {
+                    let untagged = witness.untagged_redeemer();
+                    let data = untagged.datum();
+                    let ex_units = untagged.ex_units();
+                    let index = self.mint.as_ref().and(Some(1u64)).or(Some(0)).unwrap();
```

Comment:
Are you sure about this? If yes, probably we should add a comment explaining this.

Reply:
No, I'm not. On second thought this index should be the length of assets.

Reply:
Updated.